WordPress/Classes/ViewRelated/Media/MediaLibraryViewController.swift

```diff
@@ -335,16 +342,47 @@ class MediaLibraryViewController: UIViewController {
     // MARK: - Actions
 
     @objc fileprivate func addTapped() {
+        if #available(iOS 11, *) {
+            self.showOptionsMenu()
+        }
+        else {
+            self.showMediaPicker()
+        }
+    }
+
+    private func showMediaPicker() {
         let options = WPMediaPickerOptions()
         options.showMostRecentFirst = true
         options.filter = [.all]
         let picker = WPNavigationMediaPickerViewController(options: options)
         picker.dataSource = WPPHAssetDataSource()
         picker.delegate = self
-
+        
         present(picker, animated: true, completion: nil)
     }
-
+    
+    private func showOptionsMenu() {
+        let menuAlert = UIAlertController(title: nil, message: nil, preferredStyle: UIAlertControllerStyle.actionSheet)
+        
+        menuAlert.addAction(UIAlertAction(title: NSLocalizedString("Photo Library", comment: "Menu option for selecting media from the device's photo library."), style: UIAlertActionStyle.default) {
```

Comment:
We have a couple of helpers you can use here to make things ever so slightly more concise:

`addDefaultActionWithTitle(_:handler:)`
`addCancelActionWithTitle(_:handler:)`